tests/DCPS/SetQosPartition/pubsub.cpp

```diff
@@ -0,0 +1,249 @@
+
+  MessageTypeSupportImpl::_var_type sub_type_support = new MessageTypeSupportImpl();
+
+  sub_type_support->register_type(sub_participant.in(), "");
+
+  CORBA::String_var sub_type_name = sub_type_support->get_type_name();
+
+  DDS::TopicQos sub_topic_qos;
+  sub_participant->get_default_topic_qos(sub_topic_qos);
+  DDS::Topic_var sub_topic = sub_participant->create_topic(TOPIC,
+                                                           sub_type_name.in(),
+                                                           sub_topic_qos,
+                                                           DDS::TopicListener::_nil(),
+                                                           ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  DDS::SubscriberQos sub_qos_1;
+  sub_participant->get_default_subscriber_qos(sub_qos_1);
+
+  sub_qos_1.partition.name.length(1);
+  sub_qos_1.partition.name[0] = PARTITION_A;
+
+  DDS::Subscriber_var sub_1 = sub_participant->create_subscriber(sub_qos_1, DDS::SubscriberListener::_nil(),
+                                                                 ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  DDS::SubscriberQos sub_qos_2;
+  sub_participant->get_default_subscriber_qos(sub_qos_2);
+
+  sub_qos_2.partition.name.length(1);
+  sub_qos_2.partition.name[0] = PARTITION_B;
+
+  DDS::Subscriber_var sub_2 = sub_participant->create_subscriber(sub_qos_2, DDS::SubscriberListener::_nil(),
+                                                                 ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  DDS::DataReaderQos dr_qos;
+  sub_1->get_default_datareader_qos(dr_qos);
+  dr_qos.reliability.kind = DDS::RELIABLE_RELIABILITY_QOS;
+
+  DDS::DataReaderListener_var listener_1(new DataReaderListenerImpl(dcs, SUBSCRIBER_ACTOR_1));
+
+  DDS::DataReader_var dr_1 = sub_1->create_datareader(sub_topic.in(),
+                                                      dr_qos,
+                                                      listener_1.in(),
+                                                      ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  DDS::DataReaderListener_var listener_2(new DataReaderListenerImpl(dcs, SUBSCRIBER_ACTOR_2));
+
+  DDS::DataReader_var dr_2 = sub_2->create_datareader(sub_topic.in(),
+                                                      dr_qos,
+                                                      listener_2.in(),
+                                                      ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  DataReaderListenerImpl* listener_impl_1 = dynamic_cast<DataReaderListenerImpl*>(listener_1.in());
+
+  DataReaderListenerImpl* listener_impl_2 = dynamic_cast<DataReaderListenerImpl*>(listener_2.in());
+
+  // Write to PARTITION_A, one reader.
+  ::DDS::InstanceHandle_t reader_handle_1 = -1;
+  {
+    listener_impl_1->reset(EXPECTED_READS_1, 10);
+    OpenDDS::DCPS::unique_ptr<Writer> writer(new Writer(dw.in(), " 1", 10));
+
```

Comment:
suggest change:
    OpenDDS::DCPS::unique_ptr<Writer> writer(new Writer(dw, " 1", 10));

Reply:
Could a stack variable be used here? Writer writer(dw, " 1", 10));